core/runtime/jsruntime.rs

```diff
@@ -907,59 +921,89 @@ impl JsRuntime {
     module_map.mod_evaluate_sync(scope, mod_id).unwrap();
   }
 
+  /// Executes built-in scripts and ES modules.
+  fn execute_builtin_sources(
+    &mut self,
+    realm: &JsRealm,
+    module_map: &Rc<ModuleMap>,
+    files_loaded: &mut Vec<&'static str>,
+  ) -> Result<(), Error> {
+    for file_source in &BUILTIN_SOURCES {
+      if let ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(path) =
+        file_source.code
+      {
+        files_loaded.push(path);
+      }
+
+      realm.execute_script(
+        self.v8_isolate(),
+        file_source.specifier,
+        file_source.load()?,
+      )?;
+    }
+
+    for file_source in &BUILTIN_ES_MODULES {
+      if let ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(path) =
+        file_source.code
+      {
+        files_loaded.push(path);
+      }
+      let mut scope = realm.handle_scope(self.v8_isolate());
+      module_map.lazy_load_es_module_from_code(
+        &mut scope,
+        file_source.specifier,
+        file_source.load()?,
+      )?;
+    }
+
+    Ok(())
+  }
+
   /// Initializes JS of provided Extensions in the given realm.
   fn init_extension_js(
     &mut self,
-    extensions: &mut [Extension],
-  ) -> Result<(), Error> {
+    extensions: Vec<Extension>,
+  ) -> Result<Vec<&'static str>, Error> {
     // Initialization of JS happens in phases:
     // 1. Iterate through all extensions:
     //  a. Execute all extension "script" JS files
     //  b. Load all extension "module" JS files (but do not execute them yet)
     // 2. Iterate through all extensions:
     //  a. If an extension has a `esm_entry_point`, execute it.
+    let mut files_loaded = Vec::with_capacity(128);
     let realm = JsRealm::clone(&self.inner.main_realm);
     let context_global = realm.context();
     let module_map = realm.0.module_map();
+
+    // TODO(bartlomieju): hoist it out of this method.
+    // TODO(bartlomieju): this is somewhat duplicated in `bindings::initialize_context`,
+    // but for migration period we need to have ops available in both `Deno.core.ops`
+    // as well as have them available in "virtual ops module"
+    // if !matches!(
+    //   self.init_mode,
+    //   InitMode::FromSnapshot {
+    //     skip_op_registration: true
+    //   }
+    // ) {
+    if self.init_mode == InitMode::New {
+      self.execute_virtual_ops_module(context_global, module_map.clone());
+    }
+
+    // TODO(bartlomieju): hoist it out of this method.
+    if self.init_mode == InitMode::New {
+      self.execute_builtin_sources(&realm, &module_map, &mut files_loaded)?;
+    }
+
+    // TODO(bartlomieju): hoist it out of this method.
+    self.init_cbs(&realm);
```

Comment:
All of these were deeply intertwined with execution of extension JS code - but there's no coupling between them. I might punt on further refactoring until this PR lands.